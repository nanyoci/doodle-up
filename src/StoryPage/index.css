@import url('https://fonts.googleapis.com/css2?family=Caveat+Brush&display=swap');

.story-scene-container {
	width: 100%;
	background-image: url("../assets/paper-texture.jpg");
	background-color: #d0d0d0;
	background-size: cover;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.story-scene {
	position: relative;
	width: 80%;
	margin: auto;
}

.story-bg {
	width: 100%;
}

.story-object {
	position: absolute;
    width: 30%;
    top: 50%;
    left: 60%;
}

.story-text {
	font-family: 'Caveat Brush', cursive;
	font-size: 36px;
	text-align: center;
	margin: 30px auto;
}