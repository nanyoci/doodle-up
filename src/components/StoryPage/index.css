.story-scene-container {
	width: 100%;
	background-image: url("../../assets/paper-texture.jpg");
	background-color: #d0d0d0;
	background-size: cover;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.story-scene {
	position: relative;
	width: 100%;
	max-width: 2000px;
	margin: auto;
}

.story-bg {
	width: 100%;
}

.story-object {
	position: absolute;
    width: 30%;
    top: 50%;
    left: 60%;
}

.story-text {
	font-family: 'Architects Daughter', cursive;
	font-size: 20px;
	text-align: center;
	margin-top: 40px;
	margin-bottom: 0px;
}