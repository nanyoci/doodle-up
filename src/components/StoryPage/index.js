import React from 'react';

import useSound from 'use-sound';
import drawingSample from '../../assets/drawing-sample.png';
import audioImage from './../../assets/audio.svg';
import Page from '../Page';
import './index.css';
import NextButton from '../NextButton';

// TODO: Import drawings
export default function StoryPage(props) {
	const {
		stage: {
			audio,
			description,
			image,
			drawings,
		},
		onComplete,
	} = props;

	const [playNarration] = useSound(audio);

	return (
		<Page>
			<div className="story-page-container">
				<div className="story-scene-container">
					<div className="story-scene">
						<img src={image} className="story-bg" alt="Story background" />
						{
							drawings
							&& drawings.map(drawing => (
								<img
									key={drawing.stage_id}
									src={drawingSample}
									className="story-object"
									style={{
										width: `${drawing.width / 16}%`,
										top: `${drawing.top / 9}%`,
										left: `${drawing.left / 16}%`,
									}}
									alt="Child's drawing of the story object"
								/>
							))
						}
					</div>
				</div>
				<img src={audioImage} className="audio-icon" alt="Play audio" onClick={playNarration} />
				<p className="story-text">{description}</p>
				<NextButton onClick={onComplete} />
			</div>
		</Page>
	)
}
