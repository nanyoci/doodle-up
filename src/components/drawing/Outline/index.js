import React from 'react';

import './index.css'

function Outline({ image }) {
  return (
    <div
      className="drawing-outline"
      style={{
        backgroundImage: `url("${image}")`,
      }}
    />
  );
}

export default Outline;
