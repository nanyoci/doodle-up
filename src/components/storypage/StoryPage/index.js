import React from 'react';
import useSound from 'use-sound';

import audioImage from '../../../assets/audio.svg';
import Page from '../../common/Page';
import NextButton from '../../common/NextButton';
import './index.css';

export function StoryPage(props) {
	const {
		stage: {
			audio,
			description,
			image,
			drawings,
		},
		onComplete,
		progress,
	} = props;

	const [playNarration] = useSound(audio);

	let drawingsWithUrls = [];

	if (drawings) {
		for (let i = 0; i < drawings.length; i++) {
			const {
				stage_id,
				width,
				top,
				left,
			} = drawings[i];
			
			let stageProgress = progress.stages.find(stage => stage.stage_id === stage_id);
			
			if (stageProgress && stageProgress.image_url && stageProgress.image_url !== "") {
				drawingsWithUrls.push(
					<img
						key={stage_id}
						src={stageProgress.image_url}
						className="story-object"
						style={{
							width: `${width / 16}%`,
							top: `${top / 9}%`,
							left: `${left / 16}%`,
						}}
						alt="Child's drawing of the story object"
					/>
				)
			}
		}
	}

	return (
		<Page>
			<div className="story-page-container">
				<div className="story-scene-container">
					<div className="story-scene">
						<img src={image} className="story-bg" alt="Story background" />
						{drawingsWithUrls}
					</div>
				</div>
				<img src={audioImage} className="audio-icon" alt="Play audio" onClick={playNarration} />
				<p className="story-text">{description}</p>
				<NextButton onClick={onComplete} />
			</div>
		</Page>
	)
}

export default StoryPage;
