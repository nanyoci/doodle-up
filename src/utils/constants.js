export const API_URL = 'https://doodle-up-backend.herokuapp.com';