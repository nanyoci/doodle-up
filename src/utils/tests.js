import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}