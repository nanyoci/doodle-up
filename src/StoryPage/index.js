import React from 'react';

import bg from '../assets/sample-bg.png';
import './index.css';
import drawing from '../assets/drawing-sample.png';
import Page from '../Page';

export default function StoryPage(props) {
	// const {
	// 	drawing,
	// } = props;

	return (
		<Page>
			<div className="story-scene-container">
				<div className="story-scene">
					<img src={bg} className="story-bg" alt="Story background" />
					<img src={drawing} className="story-object" alt="Child's drawing of the story object" />
				</div>
			</div>
			<p className="story-text">Once upon a time, there was a rabbit named Henry.</p>
		</Page>
	)
}
