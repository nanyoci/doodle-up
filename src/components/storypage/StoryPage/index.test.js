import React from 'react'
import { cleanup, act } from '@testing-library/react'

import audio from "./sample/audio.mp3";
import background from "./sample/background.png";
import drawing from "./sample/drawing.png";
import StoryPage from './index.js';
import { renderWithReduxRouter } from '../../../utils/tests';

jest.mock('axios');
console.warn = jest.fn()

beforeEach(() => {
	window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
})

afterEach(() => {
	cleanup();
	jest.clearAllMocks();
});

const stage = {
	audio,
	description: "The three little pigs left their mother and home to build their own houses.",
	image: background,
	stage_id: 3,
	type: "storypage",
	drawings: [
		{
			left: 900,
			stage_id: 2,
			top: 455,
			width: 200
		}
	],
};

const progress = {
	completed: false,
	stages: [
		{
			completed: true,
			stage_id: 1
		},
		{
			completed: true,
			stage_id: 2,
			image_url: drawing,
		},
	]
};

it('should take a snapshot', async () => {
	let asFragment = {};
	await act(async () => {
		({ asFragment } = renderWithReduxRouter(
			<StoryPage
				stage={stage}
				onComplete={() => { }}
				progress={progress}
			/>
		))
	});
	expect(asFragment()).toMatchSnapshot();
})
