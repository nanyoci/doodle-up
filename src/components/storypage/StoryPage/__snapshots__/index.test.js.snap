// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <audio />
    <div
      class="wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="DoodleUp"
          class="logo-corner"
          src="doodle-up.svg"
        />
      </a>
      <div
        class="story-page-container"
      >
        <div
          class="story-scene-container"
        >
          <div
            class="story-scene"
          >
            <img
              alt="Story background"
              class="story-bg"
              src="background.png"
            />
            <img
              alt="Child's drawing of the story object"
              class="story-object"
              src="drawing.png"
              style="width: 12.5%; top: 50.55555555555556%; left: 56.25%;"
            />
          </div>
        </div>
        <button
          alt="Play audio"
          class="ui violet massive circular icon button audio-icon audio-icon-story"
        >
          <i
            aria-hidden="true"
            class="volume up icon"
          />
        </button>
        <p
          class="story-text"
        >
          The three little pigs left their mother and home to build their own houses.
        </p>
        <button
          alt="next"
          class="ui yellow massive circular icon button next"
          data-testid="next-button"
        >
          <i
            aria-hidden="true"
            class="arrow right icon"
          />
        </button>
      </div>
    </div>
    <div
      class="control-buttons"
    >
      <button
        class="ui yellow massive circular icon button"
        id="signOutBtn"
      >
        <i
          aria-hidden="true"
          class="sign-out icon"
        />
      </button>
      <button
        class="ui yellow massive circular icon button"
        id="musicControlBtn"
      >
        <i
          aria-hidden="true"
          class="mute icon"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
